### PR TITLE
Dockerize collab

### PR DESCRIPTION
**Pull request by Herrekt**
Refactored naming of socket connections.
Removed excess console logs.
Added npm test (Only had one in mind since I did not make many logs/responses for connections). If there's any more can suggest and I'll make the required changes.
Dockerized Collab and tested functionality.

### CODE REVIEW

File: collaboration-service/test/test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        client2.emit("join", roomId);
+        client1.emit("change", codeChange)
+        client2.on("receive", (payload) => {
+            if (payload.code) {
+                payload.code.should.equal("print('Hello World')");
+                done();
+            }
+        })
+    })
+})
```
Comment (hiongkaihan) [inline collaboration-service/test/test.js:23]: Should add Collab Service CI along with tests, maybe can separate testing with CI in another PR

File: docker-compose.backend.yml [hunk 2]
```diff
@@ -69,6 +69,13 @@ services:
     restart: unless-stopped
     ports:
       - 8005:8005
+
+  collaboration-service:
+    container_name: collaboration-service
+    build: ./collaboration-service
+    restart: unless-stopped
+    ports:
+      - 8003:8003
```
Comment (hiongkaihan) [inline docker-compose.backend.yml:78]: Minor issue, put this between the services using port 8002 and 8004 for easier readability

File: docker-compose.yml [hunk 3]
```diff
@@ -82,5 +82,12 @@ services:
     ports:
       - 8005:8005
 
+  collaboration-service:
+    container_name: collaboration-service
+    build: ./collaboration-service
+    restart: unless-stopped
+    ports:
+      - 8003:8003
+
```
Comment (hiongkaihan) [inline docker-compose.yml:91]: Minor issue, put this between the services using port 8002 and 8004 for easier readability

File: frontend/src/components/InterviewPage.js [hunk 4]
```diff
@@ -93,9 +93,7 @@ const InterviewPage = () => {
             <div className="container mt-5">
                 <ReactMarkdown>{questionDescription}</ReactMarkdown>
             </div>
-            <div>
-                <CodeEditor socket={collabSocket} roomId={interviewId} />
-            </div>
+            <CodeEditor socket={collabSocket} roomId={interviewId} />
             <Chat socket={chatSocket} username={username} room={interviewId} />
```
Comment (hiongkaihan) [inline frontend/src/components/InterviewPage.js:97]: should standardize room variable as either room or roomId, collab socket using roomId variable but chat socket using room variable.

Comment (hiongkaihan) [reply frontend/src/components/InterviewPage.js:97]: Can check with the rest if need to change

**Review by hiongkaihan — CHANGES_REQUESTED** [review]

File: collaboration-service/test/test.js (hunk 1 above)
Comment (TheSpaceCuber) [reply collaboration-service/test/test.js:23]: Yes can separate tests + CI in another PR

**Review by TheSpaceCuber — APPROVED** [review]
Tested

**Comment by TheSpaceCuber** [comment]
@Herrekt Any update on this PR?

File: frontend/src/components/InterviewPage.js [hunk 5]
```diff
@@ -89,24 +89,22 @@ const InterviewPage = () => {
...
-                <CodeEditor socket={collabSocket} roomId={interviewId} />
-            </div>
+            <CodeEditor socket={collabSocket} room={interviewId} />
             <Chat socket={chatSocket} username={username} room={interviewId} />
-            <div className="d-flex justify-content-center mt-5">
-                <Button variant="danger" onClick={handleShowEndInterview}>End interview</Button>
-                <Modal className="deleteModal" show={showEndInterview} onHide={handleCloseEndInterview} keyboard={false} animation={false}>
+            <div className='d-flex justify-content-center mt-5'>
+                <Button variant='danger' onClick={handleShowEndInterview}>End interview</Button>
+                <Modal className='deleteModal' show={showEndInterview} onHide={handleCloseEndInterview} keyboard={false} animation={false}>
```
Comment (hiongkaihan) [inline frontend/src/components/InterviewPage.js:100]: Should revert all the changes from single quotes back to double quotes, not consistent with the rest of frontend

**Review by hiongkaihan — CHANGES_REQUESTED** [review]

**Review by TheSpaceCuber — APPROVED** [review]
Ok. Please resolve convos if they've been addressed @Herrekt ,can push for this to be merged soon as it's been quite a few days.

**Review by hiongkaihan — APPROVED** [review]

**Pull request merged**